### PR TITLE
Add Lambda implementation explainer for FeelGood

### PR DESCRIPTION
**Pull request by anishprasad01**
Covers SNS IMA (Very briefly), aliasing, env vars, and cloudwatch trigger + cron syntax, and links to docs pages or helpful resources.

### CODE REVIEW

File: _docs/feelgood.md [hunk 1]
```diff
@@ -141,9 +141,21 @@ More info coming soon.
 
 ### Lambda
 
-More info coming soon.
+AWS Lambda handles all of FeelGood's computation. Within AWS Lambda, there are three important things that must be configured in order for FeelGood's code to operate correctly.
+
+**Give Lambda access to SNS**
+    By default, creating a Lambda function from scratch, adding SNS code, and pressing "Test". won't work. Why? Simple, because the Lambda function does not have *access* to SNS. Cloud systems operate on a least-required permissions model, meaning that in order for your Lambda function to access SNS, you must explicitly grant Lambda access to SNS (and any other AWS services you might want Lambda to access). Refer to [AWS documentation on IAM permissions for SNS.](https://docs.aws.amazon.com/sns/latest/dg/sns-using-identity-based-policies.html)
```
Comment (etcadinfinitum) [inline _docs/feelgood.md]: _goooooood_. Adding a note at the end about our fancy AWS page which talks about IAM (it's currently a WIP)

```suggestion
    By default, creating a Lambda function from scratch, adding SNS code, and pressing "Test". won't work. Why? Simple, because the Lambda function does not have *access* to SNS. Cloud systems operate on a least-required permissions model, meaning that in order for your Lambda function to access SNS, you must explicitly grant Lambda access to SNS (and any other AWS services you might want Lambda to access). Refer to [AWS documentation on IAM permissions for SNS](https://docs.aws.amazon.com/sns/latest/dg/sns-using-identity-based-policies.html), and check out [the TL;DR explanation of IAM on the documentation site for this hackathon]({% link _docs/aws_secrets.md %}).
```

File: _docs/feelgood.md [hunk 2]
```diff
@@ -141,9 +141,21 @@ More info coming soon.
...
...order for FeelGood messages to be sent out at a specific time, we use AWS CloudWatch events as a trigger to run the Lambda publish function at a specific time. Select an alias in the menu that you wish to trigger. In the *Designer* pane, there is a button called **Add trigger**. Clicking on that will bring up a drop down with a number of potential triggering services. FeelGood uses *CloudWatch Events/EventBridge*. Now AWS will present a menu to configure a rule. This rule will trigger the Lambda function. **Schedule expression** is where you define the time interval that the function should be triggered at. FeelGood uses cron syntax [(a good reference on cron syntax we found useful here)](https://www.adminschoice.com/crontab-quick-reference). Cron syntax uses UTC time. In FeelGood, the trigger uses the cron syntax **cron(0 18 * * ? *)**, which means every day at 10 AM PST. We recommend looking up a converter from UTC to your timezone. Saving and enabling this trigger will run the function at the specified time. AWS has a good explainer on creating CloudWatch Events Rules [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/Create-CloudWatch-Events-Scheduled-Rule.html).
```
Comment (etcadinfinitum) [inline _docs/feelgood.md]: Changed a few things here:

* for code calls, wrap them in backticks. *cron(0 18 * * ? *)* becomes `cron(0 18 * * ? *)` and the asterisks get respected as raw characters; plus, it's a genuine API call.
* For buttons/GUI items, use consistent markup (e.g. make "Designer pane" bold instead of italics.

```suggestion
    In order for FeelGood messages to be sent out at a specific time, we use AWS CloudWatch events as a trigger to run the Lambda publish function at a specific time. Select an alias in the menu that you wish to trigger. In the **Designer** pane, there is a button called **Add trigger**. Clicking on that will bring up a drop down with a number of potential triggering services. FeelGood uses **CloudWatch Events/EventBridge**. Now AWS will present a menu to configure a rule. This rule will trigger the Lambda function. **Schedule expression** is where you define the time interval that the function should be triggered at. FeelGood uses cron syntax identical to the syntax used to schedule recurring jobs in native Unix systems; a good reference on cron syntax we found useful is available [here](https://www.adminschoice.com/crontab-quick-reference). Cron syntax uses UTC time. In FeelGood, the trigger uses the cron syntax `cron(0 18 * * ? *)`, which means every day at 10 AM PST. We recommend looking up a converter from UTC to your timezone. Saving and enabling this trigger will run the function at the specified time. AWS has a good explainer on creating CloudWatch Events Rules [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/Create-CloudWatch-Events-Scheduled-Rule.html).
```

File: _docs/feelgood.md [hunk 3]
```diff
@@ -141,9 +141,21 @@ More info coming soon.
...
...sion*, enter a description, and Publish it to create a new version. This new version will retain the functionality that it has at the time that it is published. Refer to the [AWS Docs on Lambda Versions](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html) for more detail on what you can do with versioning. Aliases allow you to refer to different versions of a function with the same name. So for example in API Gateway you can link a GET request with a Lambda function using the alias "API_GET_Version". Then, no matter what function version you connect to that alias, be it 1, 10, or 1000, you will not need to change the API Gateway configuration when you update the Lambda code. You do however, need to modify which function version the alias will point to each time you publish a new version. You *can* alias to the $LATEST version in order to have Lambda automatically use whatever code exists when you Save the function, but this is not recommended. You should only alias to function versions that you have tested and verified to work. Further documentation on function aliases is available [here](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html).
```
Comment (etcadinfinitum) [inline _docs/feelgood.md]: Recommend avoiding referring to internal Lambda features as "functions"; the terminology could get confusing quickly for us and for new cloud users.

```suggestion
    Lambda has a feature called **Aliases**. Aliases are used in combination with function versions to manage what version of the Lambda function will be called by other AWS services. In FeelGood, aliases are used to connect API Gateway endpoints to the publish and subscribe Lambda functions. Under the **Actions** menu in the function, click on **Publish new version**, enter a description, and publish it to create a new version. This new version will retain the functionality that it has at the time that it is published. Refer to the [AWS Docs on Lambda Versions](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html) for more detail on what you can do with versioning. Aliases allow you to refer to different versions of a function with the same name. So for example in API Gateway you can link a GET request with a Lambda function using the alias "API_GET_Version". Then, no matter what function version you connect to that alias, be it 1, 10, or 1000, you will not need to change the API Gateway configuration when you update the Lambda code. You do however, need to modify which function version the alias will point to each time you publish a new version. You *can* alias to the `$LATEST` version in order to have Lambda automatically use whatever code exists when you Save the function, but this is not recommended. You should only alias to function versions that you have tested and verified to work. Further documentation on function aliases is available [here](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html).
```

**Review by etcadinfinitum — CHANGES_REQUESTED** [review]
Overall, this looks great. I added some nitpicky comments which IMO should be incorporated, and have some advice. 2 big thoughts about this:

1. Break up each paragraph into smaller paragraphs (or perhaps bullets). Wall-of-text is always hard to parse, and there's a lot of good info here. The existing content is well-organized, and it would be easy to split the large text blocks into bite-sized paragraphs without rearranging content too much. Bold the first sentence/phrase which you think is key to the point you're trying to make.
2. Use headings instead of bolded sentences -- _especially if you break up each paragraph_. You can use an H4 `####` heading if you like.

**Comment by redpandarevolt** [comment]
This is very thoroughly written. As a new user to cloud computing I was looking to see if given the project description did I feel like I can take this on and build it. The answer is yes. The formatting is well explained and the links to outside resources. It flows nicely. Very good work.

File: _docs/feelgood.md [hunk 4]
```diff
@@ -141,13 +141,33 @@ More info coming soon.
...
+
+#### Set up aliases for Lambda functions
+- Lambda has a feature called **Aliases**. Aliases are used in combination with function versions to manage what version of the Lambda function will be called by other AWS services. In FeelGood, aliases are used to connect API Gateway endpoints to the publish and subscribe Lambda functions. 
+
+- In order to create and alias, you must first have a **published function version** to connect it to. Under the Actions menu in the function, click on **Publish new version**, enter a description, and publish it to create a new version. This new version will retain the functionality that it has at the time that it is published. Refer to the [AWS Docs on Lambda Versions](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html) for more detail on what you can do with versioning.
```
Comment (huyn8) [inline _docs/feelgood.md:157]:
```suggestion
- In order to create an alias, you must first have a **published function version** to connect it to. 
```

Comment (huyn8) [inline _docs/feelgood.md:157]:
```suggestion
- In order to create and alias, you must first have a **published function version** to connect it to. Under the Actions menu in the function, click on **Publish new version**, enter a description, and publish it to create a new version. This new version will retain the functionality that it has at the time that it is published. Refer to the [AWS Docs on Lambda Versions](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html) for more detail on what you can do with versioning.
```

Comment (huyn8) [inline _docs/feelgood.md]:
```suggestion
- In order to create an alias, you must first have a **published function version** to connect it to. Under the Actions menu in the function, click on **Publish new version**, enter a description, and publish it to create a new version. This new version will retain the functionality that it has at the time that it is published. Refer to the [AWS Docs on Lambda Versions](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html) for more detail on what you can do with versioning.
```

File: _docs/feelgood.md [hunk 5]
```diff
@@ -141,13 +141,33 @@ More info coming soon.
...
+    
+#### Configure CloudWatch Trigger Invocation
+- In order for FeelGood messages to be sent out at a specific time, we use AWS CloudWatch Events/EventBridge as a trigger to run the Lambda publish function at a specific time. 
+
+- First, select an alias for in the menu for the Lambda function that you wish to trigger. In the **Designer** pane, there is a button called **Add trigger**. Clicking on that will bring up a drop down with a number of potential triggering services.
```
Comment (huyn8) [inline _docs/feelgood.md]:
```suggestion
- First, select an alias in the menu for the Lambda function that you wish to trigger. In the **Designer** pane, there is a button called **Add trigger**. Clicking on that will bring up a drop-down with a number of potential triggering services. 
```

**Review by huyn8 — COMMENTED** [review]
I'm not an expert on this content-wise so I was just checking on some typo.

**Review by etcadinfinitum — APPROVED** [review]
LGTM!
